public/app/plugins/panel/heatmap/fields.ts

```diff
@@ -98,36 +103,32 @@ export function prepareHeatmapData({
   });
 
   if (options.calculate) {
+    // if calculate is true, we need to have the default values for the calculation if they don't exist
+    let calculation = getCalculationObjectWithDefaults(options.calculation);
+    calculation =
+      calculation.xBuckets?.valid && calculation.yBuckets?.valid ? calculation : getCalculationObjectWithDefaults();
```

Comment:
When a value is not defined, the calculation has a fallback 

https://github.com/grafana/grafana/blob/main/public/app/features/transformers/calculateHeatmap/heatmap.ts#L507

This was working before because the old duration calculation would error out and return undefined with milliseconds. The new calculation returns valid millisecond values more, so we need to set a validity and use that to know when to send no data and use the fallback.